pkg/controller/mizar/mizar-pod-controller.go

```diff
@@ -191,6 +207,82 @@ func (c *MizarPodController) handle(keyWithEventType KeyWithEventType) error {
 		}
 	}
 
+	//The annotations of vpc and subnet should not be added into pods
+	//which use host networking
+	if eventType == EventType_Create && !obj.Spec.HostNetwork {
+		klog.V(4).Infof("Get hostIP (%s) - podIP(%s)", obj.Status.HostIP, obj.Status.PodIP)
+		network, err := c.netLister.NetworksWithMultiTenancy(tenant).Get(defaultNetworkName)
+
+		if err != nil {
+			klog.Warningf("Failed to retrieve network in local cache by tenant %s, name %s: %v", tenant, defaultNetworkName, err)
+			return err
+		}
+		klog.Infof("Get network: %#v.", network)
+
+		if network.Spec.Type != mizarNetworkType || network.Status.Phase != arktosextv1.NetworkReady {
+			klog.Warningf("The arktos network %s is not mizar type or is not Ready.", network.Name)
+			return nil
+		}
+		klog.V(4).Infof("Get network %s - VPCID: %s.", network.Name, network.Spec.VPCID)
+
+		vpc := network.Spec.VPCID
+		if len(vpc) == 0 {
+			//vpc = tenant + vpcSuffix
+			vpc = fmt.Sprintf("%s%s", tenant, vpcSuffix)
+		}
+		//subnet := vpc + subnetSuffix
+		subnet := fmt.Sprintf("%s%s", vpc, subnetSuffix)
+
+		needUpdate := false
```

Comment:
Better extract the calculation of needUpdate target object to another function. Make the handle function simple and easier to read. Also don't use excessive log for condition checking. Even though the log level is not high, it still takes CPU time.

Reply:
will make change for master branch

Reply:
Done and updated in q131172019:CarlXie-poc-2022-01-30-tenant-vpc-support-final